src/base/bittorrent/session.cpp

```diff
@@ -1837,13 +1837,16 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
         QString rootPath = torrent->rootPath(true);
         if (!rootPath.isEmpty())
             // torrent with root folder
-            m_savePathsToRemove[torrent->hash()] = rootPath;
+            m_removingTorrents[torrent->hash()] = {torrent->name(), rootPath, deleteLocalFiles};
```

Comment:
@sledgehammer999:  My apologies for taking so long to come up with a good name… 

`m_torrentsBeingRemoved` better describes the behavior of this.

Reply:
Yes, I knew that "being" had to be involved. But I had a brainfart and couldn't imaging the correct word sequence. `m_torrentsBeingRemoved` and `TorrentDataBeingRemoved` should be nice (although a little long)

Reply:
>TorrentDataBeingRemoved

Actually, we can do without a separate structure, but simply use `QVariantHash` to store needed data. Or, even better, `QHash<int, QVariant>` with some predefined constants for its keys (or, even maybe, something like std::tuple).

Reply:
>Actually, we can do without a separate structure, but simply use QVariantHash to store needed data. Or, even better, QHash<int, QVariant> with some predefined constants for its keys (or, even maybe, something like std::tuple).

I can't imagine a solution using the above. How do we save the 3 things tracked by the current struct?

Reply:
Oh now I understood. You want to store the 3 tracked things inside a QHash. Each of the 3 having a predefined constant id. Then store that QHash inside another QHash tracking it by the infohash.
To me it seems a bit more complex and unintuitive that way.

Reply:
Okay, forget it. These are just some ideas. 

As for names, I think we should not seek "perfect" English there (in contrast to user-visible strings). We need to use some limited subset of the language that will convey the desired meaning, but will not be too excessive. So "RemovingTorrentData" is quite appropriate here.